src/menu/Menu.ts

```diff
@@ -0,0 +1,212 @@
+import { createHandle } from '@dojo/core/lang';
+import uuid from '@dojo/core/uuid';
+import { v } from '@dojo/widget-core/d';
+import { DNode } from '@dojo/widget-core/interfaces';
+import ThemeableMixin, { theme, ThemeableProperties } from '@dojo/widget-core/mixins/Themeable';
+import WidgetBase from '@dojo/widget-core/WidgetBase';
+import * as css from './styles/menu.m.css';
+
+export type Orientation = 'horizontal' | 'vertical';
+
+export type Role = 'menu' | 'menubar';
+
+/**
+ * @type MenuProperties
+ *
+ * Properties that can be set on a Menu component.
+ *
+ * @property active				Determines whether the menu has focus.
+ * @property activeIndex		Determines the index of the focused item.
+ * @property disabled			Determines whether the menu is disabled.
+ * @property id					The widget ID. Defaults to a random string.
+ * @property nested				Indicates whether the menu is nested within another menu.
```

Comment:
It is only used for styling, and since `SubMenu` renders its menu to a container node it can control when animating, and since there are other classes for positioning, this is probably unnecessary. I'll remove it.